packages/flutter_tools/lib/src/ios/xcresult.dart

```diff
@@ -40,30 +40,37 @@ class XCResultGenerator {
     List<XCResultIssueDiscarder> issueDiscarders = const <XCResultIssueDiscarder>[],
   }) async {
     final Version? xcodeVersion = xcode.currentVersion;
-    final RunResult result = await processUtils.run(<String>[
-      ...xcode.xcrunCommand(),
-      'xcresulttool',
-      'get',
-      // See https://github.com/flutter/flutter/issues/151502
-      if (xcodeVersion != null && xcodeVersion >= Version(16, 0, 0)) '--legacy',
-      '--path',
-      resultPath,
-      '--format',
-      'json',
-    ]);
+    final bool useNewCommand = xcodeVersion != null && xcodeVersion >= Version(16, 0, 0);
+
+    // Base command for xcrun
+    final command = <String>[...xcode.xcrunCommand(), 'xcresulttool'];
+
+    // Determine the correct subcommand and arguments based on the Xcode version.
+    if (useNewCommand) {
+      // For Xcode 16+, the 'get' command requires a specific subcommand.
+      // 'build-results' is the correct choice for parsing compilation/build errors.
+      command.addAll(<String>['get', 'build-results', '--path', resultPath, '--format', 'json']);
+    } else {
+      // For older versions of Xcode, the original 'get' command works directly.
+      command.addAll(<String>['get', '--path', resultPath, '--format', 'json']);
+    }
+
+    // Run the constructed command.
```

Comment:
For example, this comment can also be removed -  the actual code makes it obvious its running a command.